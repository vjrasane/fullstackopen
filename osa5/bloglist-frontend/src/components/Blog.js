import React from 'react'
import Expandable from './Expandable'
import BlogDetails from './BlogDetails'
import PropTypes from 'prop-types'

class Blog extends React.Component {
  toggle = () => {
    this.toggleable.toggle()
  }

  render() {
    return (
      <div className='blog'>
        <div
          className='title'
          onClick={this.toggle}>
          {this.props.blog.title}, {this.props.blog.author}
        </div>
        <Expandable ref={c => this.toggleable = c}>
          <BlogDetails
            blog={this.props.blog}
            like={this.props.like}
            remove={this.props.remove}
            username={this.props.username}
          />
        </Expandable>
      </div>
    )
  }
}

Blog.propTypes = {
  blog: PropTypes.object.isRequired,
  like: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
  username: PropTypes.string
}

export default Blog
