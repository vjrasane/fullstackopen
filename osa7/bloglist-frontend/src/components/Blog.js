import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

class Blog extends React.Component {
  render() {
    return (
      <div className='blog'>
        <div className='title'>
          <Link to={`/blogs/${this.props.blog._id}`}>{this.props.blog.title}, {this.props.blog.author}</Link>
        </div>
      </div>
    )
  }
}

Blog.propTypes = {
  blog: PropTypes.object.isRequired
}

export default Blog
