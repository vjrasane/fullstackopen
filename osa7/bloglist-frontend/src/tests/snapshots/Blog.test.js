import React from 'react'
// import jest from 'jest-mock'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import Blog from '../../components/Blog'

describe('<Blog />', () => {
  let blogObj

  beforeEach(() => {
    blogObj = {
      title: 'Simple blog title',
      author: 'Teppo Testaaja',
      url: 'http://test.url.com',
      user: 'Teppo Testaaja',
      likes: 42
    }
  })

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Blog
            blog={blogObj}/>
        </Router>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
