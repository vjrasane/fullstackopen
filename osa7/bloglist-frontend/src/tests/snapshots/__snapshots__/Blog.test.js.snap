// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog /> renders correctly 1`] = `
<div
  className="blog"
>
  <div
    className="title"
  >
    <a
      href="/blogs/undefined"
      onClick={[Function]}
    >
      Simple blog title
      , 
      Teppo Testaaja
    </a>
  </div>
</div>
`;
